### ISSUE TITLE
support for eVic Primo

### ISSUE THREAD

**Issue by Cantpickrightusername** [maelstrom2001/ArcticFox#106]
Do you plan to add eVic Primo among suported devices ?

U guys are doing great job. 
Thaks for all vapers and good luck.

**Comment by TBXin** [comment]
Hello,

Yes, we will add support for this kind of devices, but at this point we are waiting for the official firmware release.

**Comment by maelstrom2001** [comment]
Closed until official release.

**Comment by ozomax1997** [comment]
I apologise

**Comment by maelstrom2001** [comment]
Testers are welcome. Do not forget update NToolbox to the latest [nightly version](https://ci.appveyor.com/project/TBXin/NFirmwareEditor/branch/master/artifacts).
[af_170214.zip](https://github.com/maelstrom2001/ArcticFox/files/776645/af_170214.zip)

**Comment by ozomax1997** [comment]
Tried out the firmware. Mod seems to het a little warmer on usb chargeing. Also is there supposed to be a screen off animation? Dont think my rx200s had it but not able to get my hands on it atm. Been using it for about a hour now and had no issues so far otherwise

**Comment by maelstrom2001** [comment]
@ozomax1997 thanks for response! If it possible, can you compare box temperature with stock firmware while charging? I think that moderate heating is normal, especially that the charging current is 1.5 amps (on older boxes current is limited to 1.0 amp). But just in case it is useful to check :)
And yes, power off animation is implemented on all builds since 170210.

**Comment by bjoernx** [comment]
I'm on it too now. on Linux (arch/Antergos). 
first I tried to upload with evic-usb (with --no-verify aprom option, dunno, no instructions found). thought I had bricked device, box remained totally unresponsive after that...
each time removing batteries and reconnecting helped, until then pc would not recognize no device. reflashed stock 4.05 with evic-usb, works! (pheewww.... there's my vape again, at last :D) 
managed to get a connection with w7/32bit in virtualbox (forgot to add me to vboxusers group first). unresponsive after flashing with the toolbox, again. same procedure with evic-usb (on the host). 
device was then recognized in the toolbox as some 0 firmware version thing in LDR (?) boot-mode ... don't remember, sorry. flashed with arctic, this time successful.
request: please add installation procedures with evic-usb on linux! :P 
only thing I have found so far is the single battery indicator, although there are two inside (similar to somthing I read with the Dual in another thread, maybe). but actually I really don't mind that, seeing two batteries makes little sense actually. otherwise really like what I see, will try this in more detail soon and come back at you, if something pops up.

**Comment by nikosma** [comment]
Everything seems to be working properly, but i can see only one battery icon instead of two (anyway doesn't matter at all).
Excellent job!!! Thank you very much!!!!

**Comment by bjoernx** [comment]
I second nikosma, the firmware is awesome! the more I use it, the more I like. :)

I found some annoyances with battery charging and depletion, but this will be more general. 
It's the lack of hysteresis both when batteries are depleted and in the fully charged status. This has the effect when charging, that there is no timeout. Lios should be charged for no longer than 4 hours max AFAIK. Charging should stop if either 100% are reached and voltages have not dropped below - say - 4V (fore example) - reset the flag then. 
On the other end I saw the device coming in and out of low battery status a few times when depleted, because the batts reload themselves a bit when empty. So there should be some hysteresis to pin the empty status as well. Also the low/empty-level maybe should be raised, since with the voltage drop voltage goes down as low as 2.8 V - I dunno if this is safe.

I guess I should open an extra Issue for all that, what do you think?

**Comment by bjoernx** [comment]
Allright, forget what I said about voltage drops below 2.8 being dangerous. I mixed up with the specs of some 18650 I ripped from an old laptop pack a while ago (with < 1C current output), I wasn't aware that these high current things often have a much lower cut off voltage... XD the vtc5 I have here have 2.5 or 2 V, older ones have 3 (typically, afair).
Still, what I said about hysteresis would be nice to have... possibly. And the maximum charging time and avoiding trickle charging...? not sure.

**Comment by akazuk** [comment]
Hi Guys. I've tried to update with NFirmwareEditor-v6.5.43 but it says " Opened firmware file is not suitable for the connected device. ". I've tried things like connecting/reconnecting usb, taking batteries out, etc. 
Sorry ... noob here. Can anyone help ? Also in the field Device Name it says Unknown device. 

My firmware versions is 4.05 hardware version 1.0

**Comment by TBXin** [comment]
@akazuk 

Hello,

Are you trying to install nightly build of the ArcticFox firmware that was attached in this issue above?

**Comment by akazuk** [comment]
Yes, lol idk what happened but it worked now. Thanks anyway :)

**Comment by sarathmenon** [comment]
Is there a more recent firmware? I tried the one linked in here, but on flashing, the device won't boot. I had to restore stock to get it working again. I am using stock windows 7 on a virtualbox VM, with the usb device added to the VM.

**Comment by maelstrom2001** [comment]
The most recent firmware is located [here](https://github.com/maelstrom2001/ArcticFox/wiki/Changelog), also there you can find the link to nightly NToolbox which is required to install the firmware.

Please take a look on https://github.com/maelstrom2001/ArcticFox/issues/106#issuecomment-280884255
If you're using VM, box after flashing can stuck in LDROM boot mode (see status in Firmware Updater). Just flash the firmware .bin again (if you sure that it is not corrupted) or switch the box to APROM boot mode manually:
![](http://i.imgur.com/0E66kej.png)

**Comment by sarathmenon** [comment]
@maelstrom2001 Thanks for the comment. I did read through all the comments in here, but somehow missed #106. I am using the nightly NToolbox, but after flashing the latest firmware, I still get the same thing. The screen stays off, and NToolbox does not recognize the device to allow me to change the boot mode or flash the firmware again. I am on OSX, which shouldn't matter that much in the grand scheme of things. runnning NToolbox with wine gives me the same thing. I have a linux host that I might be able to retry this on.

**Comment by espezetka** [comment]
Hi guys, at first let me please say warm Thank you for your work for us. 
And if i may ask you. How does it look with full support for Primo? Any eta of release? 
Now i am using v. 170222 and everything is working fine, only one thing what is not actually acurate is battery meter. The Primo is charging only to 4.13V each battery(LG HG2 or VTC4, paired batteries) . I haven't it measured by voltmeter so it might be only a number. 
Thank you again for answer and have a nice day. ^_^

**Comment by maelstrom2001** [comment]
@espezetka thank you!
Developing is in the active phase, ATM we can't say when there will be a release, too much work remains to be done. We looking for feedback on nightly builds, 'cause we have limited number of supported boxes and we can not test everything physically :)
If you want, you can test attached build. Charging routine was modified slightly, need to test it and check that the charging on this build goes through all stages correctly. Any help in testing and responses are highly appreciated.

[af_170318.zip](https://github.com/maelstrom2001/ArcticFox/files/852716/af_170318.zip)

**Comment by espezetka** [comment]
Thank you. I will charge now and once the charging will be complete i will report.  I will try it twice with mentioned batteries and let you know

**Comment by espezetka** [comment]
Ok, LG hg2 now ended  charging.  The highest voltage i got was 4.15 and 4.14 volts. Then, still plugged into wall adapter, voltage dropped to 4.12 and 4.11 volts without any firing etc. Tomorrow i will try charging with VTC4 and with device monitor recording and will upload a csv file here.

**Comment by maelstrom2001** [comment]
> with device monitor recording and will upload a csv file here

It will be very useful :+1: suspect that while charging there is spikes in battery measurements occasionally, i. e. wrong measured and overestimated voltage cuts charging off. Monitor data at the end of charge will slightly clarify the situation. Will see :) thank you!

**Comment by espezetka** [comment]
Lines 84-87 there you can see the voltage drop when charging. Device was still plugged in and nothing happened since the drop. Now i am tryinng VTC4 so i will upload a next file to this comment once the charging is complete.
[primo lg hg2.zip](https://github.com/maelstrom2001/ArcticFox/files/853067/primo.lg.hg2.zip)

**Comment by espezetka** [comment]
Well so i changed to original firmware from joyetech and here is the result so i can say that the hardware is all right. 
![20170321_192247-1](https://cloud.githubusercontent.com/assets/26419510/24164099/f0c10ad0-0e6c-11e7-91a9-50d5bf98da17.jpg)

Other devices with ArcticFox will charge to 4.19 without any problems, but theese are a single or built in batteries. Istick pico and ipower 80w.

**Comment by maelstrom2001** [comment]
@espezetka thanks for report, working on it :)

**Comment by espezetka** [comment]
For sure mate :)

**Comment by maelstrom2001** [comment]
@espezetka can you test attached build please? Make some changes in charging routine, maybe "4.12 volts bug" will disappear :) Just in case, please do not leave mod overnight and monitor it time at time to avoid overcharging. It should stop and turn screen off at 4.19..4.23 volts depending on the voltages difference.

[af_170324.zip](https://github.com/maelstrom2001/ArcticFox/files/868961/af_170324.zip)

**Comment by espezetka** [comment]
Ok no problem, i will take a look on it tomorrow. 
Now my batteries are on 3.6V both so to 4.something V i will be on wall adaptor and then i will swich to pc  USB and start recording trhu devices monitor and Post a csv file here :)

**Comment by espezetka** [comment]
So dear Mr. Maelstrom2001 here are results after charging.
4.19 and 4.21 V for me looks good but i have to point out the board temperature. From 44 celsius on older build to 36 celsius on new one? Nice :) 
For me, this version is much better than was 170318. 
Batteries are full and the PCB is not exposed to high temperature while charging. What else to wish?
 TFR with same curve as i have used on older version, seems to work better on new version, but it might be only my opinion because i do not know if you was doing something with TFR functions too.
For me, this version is just a top notch for Primo :+1: :100: 
[LGHG2 new.zip](https://github.com/maelstrom2001/ArcticFox/files/869892/LGHG2.new.zip)

**Comment by maelstrom2001** [comment]
@espezetka thanks a lot for testing and response!
I see in attached CSV that charging was started with 4.13 volts, right? Can I ask you to make the last test? :)
Please discharge batteries to approx. 4.0 volts and put box on charge. No need to record CSV this time, just make sure that charging will continue when voltages will reach 4.12 volts (transition to 300 mA charging current, switching to the last charging stage with balancing). Thank you!

**Comment by espezetka** [comment]
Ok i will swich atty to be able to use higher wattage and faster Fischerem and will report soon. Will use wall adaptor 2.1A

**Comment by espezetka** [comment]
Discharged to 4.01V both, charging slowed down since 4.12/4.13V and continued to final 4.19/4.20V and then screen shut off. So far so good for me. :+1: 
![20170325_150334](https://cloud.githubusercontent.com/assets/26419510/24322970/064f67ee-116e-11e7-8574-8f5546dbed18.jpg)

**Comment by maelstrom2001** [comment]
![](https://cdn.meme.am/instances/400x/33617404.jpg)

**Comment by espezetka** [comment]
![the-great-gatsby-cocktails](https://cloud.githubusercontent.com/assets/26419510/24323114/4412e026-1171-11e7-87c8-58b940bdacf6.jpg)
:1st_place_medal:

**Comment by andreivoinea93** [comment]
I just tried this and it fried my evic. -.- Seriously. Smells burnt and won't start

**Comment by espezetka** [comment]
Any more info? Battery type?

**Comment by andreivoinea93** [comment]
VTC6 charged at 80% (as shown by Smok Alien)

**Comment by espezetka** [comment]
Were the batteries paired? Same age, cycles etc? Was Evic hot before it died?

**Comment by andreivoinea93** [comment]
The batteries never seen different mods, they were always together in the same mod, either my Primo or my Alien. Board temp during charging first was 50C and then cool off to 47. I took out the charger, plugged it back in, the screen started flashing like it was in a bootloop and then it died.

**Comment by espezetka** [comment]
Did you tried to insert batteries again without charging?

**Comment by andreivoinea93** [comment]
sure, nothing happens.

**Comment by espezetka** [comment]
So let's wait till Mr. Maelstrom2001 take a look on this thread. Maybe it was HW problem.

**Comment by andreivoinea93** [comment]
I doubt. It would be the too much of a coincidence.

**Comment by espezetka** [comment]
Well try this. Download official FW from joyetech page, try to flash it into mod trhu official updater. I know about similar issue but there was no smell of burning. Maybe it helps

**Comment by andreivoinea93** [comment]
It would be helpful to know if based on the software change the problem could be a blown fuse, so I can fix that myself. Otherwise I have to return it

**Comment by andreivoinea93** [comment]
I'm running linux so I can use lsusb to check if any device is alive on USB. It shows nothing. It's dead

**Comment by espezetka** [comment]
give it a try, if you will be able to put original fw back you can return it without breaking a warranty

**Comment by andreivoinea93** [comment]
Great! Joyetech website is down now, rofl.

I'll give it a try but there is no way it's gonna work :)

**Comment by TBXin** [comment]
@andreivoinea93 did you try connect device to PC without batteries and holding "plus" button?

1. Remove batteries
2. Press "plus" button and hold
3. Connect to PC
4. Release "plus" button

After this try to find device via libusb. If you find it - so, you can install any firmware.

**Comment by espezetka** [comment]
[joyetech.zip](https://github.com/maelstrom2001/ArcticFox/files/870176/joyetech.zip)
here is the original FW and updater for primo

**Comment by andreivoinea93** [comment]
Nope, no response with the plus button. Needless to say, joyetech updater doesn't see a device.

**Comment by espezetka** [comment]
This is really srtrange... Mine works fine, now charging again just for try

**Comment by maelstrom2001** [comment]
> I took out the charger, plugged it back in, the screen started flashing like it was in a bootloop and then it died.

> joyetech updater doesn't see a device.

Looks like MCU or its voltage regulator died on USB connection, really HW fault. Sad, but nothing supernatural. Remember how many were RX2/3 dead - black displays, burnt DC/DC converters etc.
Posted yesterday on Russian forum: http://www.ecigtalk.ru/forum/f17/t141141-post10289363.html#post10289363
"Burned. It was the first my device which supports USB charging, and seems exactly that was the problem".

**Comment by espezetka** [comment]
Few chargings later : still charging fine, without any problem. Device is still around 35°C when plugged into wall adaptor. Once it reaches 4.12/4.13 it slows down. 

Just reporting :)

**Comment by maelstrom2001** [comment]
@espezetka Do you know what is the most amusing? This is what the builds 170324 and 170318 are identical on MCU side, checked with disassembler :) I have a suspicion that charging is depend on batteries voltages difference. Now you have perfectly balanced cells, and charging works very smoothly. But the same build refused to work correctly on RX300 with a slightly imbalanced cells. And finally: charging routines in stock Primo and RX300 firmwares are identical. At the moment I am in a slight bewilderment :)

**Comment by espezetka** [comment]
About batteries, every monday i am putting them into external charger so maybe this is the reason for balanced cells. I recomend this to everyone, your mod might have balancer (primo has)  but external charger is a basic device for vaper. 

Maybe died primo was just a bad luck, it just happens. Sensitus was not first or last with hardware failure i think.

**Comment by OlafKroll** [comment]
170318 works fine on my Primo

**Comment by RX300N** [comment]
170331 - latest I know about - works fine both for Primo and RX300

**Comment by Pete103** [comment]
170331 works fine on my Primo too with one issue being the clock being far from accurate. My impression is Primo has an RTC chip, is this correct?

**Comment by skorpyx** [comment]
still reports "atomizer short error" if you reinstall back to official firmware so everything is fine. What is the solution?(170331) Thanks

**Comment by maelstrom2001** [comment]
@skorpyx please provide more information about your setup. Have Primo (early revision) but can not reproduce and confirm this issue on any of my atties.

**Comment by skorpyx** [comment]
Evic Primo(1)
170331, 
SMOK TFV12.. Head V12-T12 and  V12-X4 

You need more information?

**Comment by espezetka** [comment]
TFV line of atomizers have had issues with coils.. Take a look it there is connection between head and base of atty

**Comment by skorpyx** [comment]
I installed back the official rom and the error disappeared. again I installed the Arctic Fox. everything is OK. I'll try to test everything

**Comment by OlafKroll** [comment]
ähm.... 
"Atomizer short error"
The Primo in the first generation has problems with the 510er.
This problem has been fixed in the Evic Primo 2.0

**Comment by RX300N** [comment]
In fact this HW problem was fixed earlier before PRIMO 2 was released. |First batches were affected heavily (probably not all of them) but when first reviews came out and there was a lot of grunging re Joyetech 510 pin - company reacts rather swiftly. From that time on there is no problem (AFAIK). Then Primo 2 came out. The way how to distinguish if you have failing old 510 or new (still talking about Primo 1) was display color tint - some reviewers said. Updated version has slight blue tint compared to original (slight yellowish). However I am not sure if this is true - I have yellowish but from march production and 510pin is on par with RX300 - flawless.
There are still options one can try even having Primo from first batches. 
Carefully clean the pin (atomizer+mod)
Try several times to push the center pin (mod) - just to check if it is not too stiff
Try using thin (up to 1 mm) distance ring between mod top plate and atomizer bottom surface.
Try to screw the atty harder (beware not too hard) or loosen it a bit (but the connection has to be maintained)
Unscrew the atty completely wait 1-2 secs and screw it back with reasonable force.

I have had small issues with some of my Joyeteech pins (IStick 100, IStick 200, RX2/3) and different attys TFV4, Smok RTA, Crius plus and Ijoy limitless)  -  however not at all with Reux. And luckily I found my way with all of them how to operate them reliabily. None so far with RX300 a IPrimo 1 (late batch).
Just do not forget this is hardware connection issue not firmware. Even without firmware change you can find that trying options mentioned above the mod will start working with no hassles.

**Comment by OlafKroll** [comment]
read on Joytech Site

>  Applicable for various tanks
> The exquisite spring loaded 510 connector of the eVic Primo 2.0 is applicable for tanks with 510 length 
> ranged at 4.0mm to 5.5mm.Tanks are also free to go up to 25mm diameter as you prefer.

**Comment by RX300N** [comment]
One more point with Primo 1. Atomizer when reasonably tightened/screwed should be perfectly flush on top of mod. Having a small gap between mod and atty is a sign that old 510 design is used (ISticks, RX2/3.

**Comment by skorpyx** [comment]
Thank you for the advice. So far everything works.

**Comment by DrZeck** [comment]
Hi, i am also using 331 version on primo without any issues so far, works perfect, i am running 0.35 ohm SS setup with TC. beside primo i have same version on istick pico, evic basic and RX300. the only problems i had on primo only is atomizer short and tcr errors when tcr check is off, when i have enable it i did not get that message at all ... if i can help somehow in development let me know. Cheers and Great work ...

**Comment by pannal** [comment]
So this is compatible with the Primo 2.0 also?

**Comment by DrZeck** [comment]
My primo is not 2.0 but that should be same hardware ...

**Comment by Delanoss** [comment]
First of all, thanks for the hard work. In my Primo, I can not go over 300F because it gets too hot. With VTC mini, Eleaf TC100 and Cuboid Mini, works fine in 400/420F. I tested  all mods with Subtank, SS, AWG28, 3.0 ID, 0.50 ohms, smaller hole (MTL). Profile material TFR 316L.

**Comment by maelstrom2001** [comment]
@Delanoss check locked/saved atomizer resistance and compare it with other boxes. Run device monitor, for example, on Cuboid Mini, look at Live Resistance and Temperature charts while firing, then do the same on Primo. Compare resistance reading and tune locked/saved resistance for current profile, or adjust shunt value to get measurements equals to Cuboid Mini.

**Comment by Delanoss** [comment]
VTC Mini ![tvc mini](https://cloud.githubusercontent.com/assets/28121799/25546223/854c32cc-2c38-11e7-919c-00a2f3314917.jpg)

**Comment by Delanoss** [comment]
Primo 1 ![primo 1](https://cloud.githubusercontent.com/assets/28121799/25546258/9d467fcc-2c38-11e7-911c-c778424d9955.jpg)

**Comment by Delanoss** [comment]
I'm sorry @maelstrom2001 . I'm noob. I hope these help.

**Comment by nicolaisueper** [comment]
Does this work for the v2.0 of the primo too? I really dont know the difference except the max watts.

**Comment by TBXin** [comment]
@nicolaisueper hardware is different. Primo v2.0 support will be added in the upcoming release.

**Comment by nicolaisueper** [comment]
Alright. Thanks a lot for your work! If you need someone who tests the firmware before release let me know. :)

**Comment by maelstrom2001** [comment]
Released.
